apps/arclight/src/app/videoPlayerUrl/VideoPlayer.tsx

```
@@ -41,97 +91,884 @@ export function VideoPlayer({
+    const endTimeMs =
+      effectiveEndTime === Infinity ? duration : effectiveEndTime * 1000
+
+    if (currentTime < startTimeMs) return 0
+    if (currentTime > endTimeMs) return 100
+
+    const segmentDurationMs = endTimeMs - startTimeMs
+
+    // Handle edge cases to prevent NaN, Infinity or undefined values
+    if (
+      !segmentDurationMs ||
+      !isFinite(segmentDurationMs) ||
```

Biome (1.9.4)

[error] isFinite is unsafe. It attempts a type coercion. Use Number.isFinite instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isFinite instead.


(lint/suspicious/noGlobalIsFinite)